Use hub pr list

ataylorme opened the pull request:
The logic in`checkExisting` is checking _issues_, not _pull request_, to determine if there is an existing update open.

This is causing the function to return `false` when there are, indeed, open CLU pull request.

Changing to `hub pr list` instead of `hub issue` will sort this out.

`--state=open` ensures we receive only open pull requests.

`--format="%t%n"` sets the output to simple the title of the pull request, which the regex expects.

Example:
```sh
andrewtaylor:example-wordpress-composer/ (limit-clu*) $ hub pr list --format="%t%n" --state=open                                                                            
Update Composer dependencies (2019-05-24-00-02)
BT-14: Bitbucket Pipelines preliminary configuration
Update Composer dependencies (2018-12-06-22-44)
Adds comments on how to include custom plugins/themes in the repo
Update ReadMe
Edits
Use `composer.lock` to generate the `composer-cache` key
WP & phpcs version bump
Bumping WP version to 4.9
Updated readme with a link to the installation
```

`hub pr list` was added in [hub version `2.3.0`](https://github.com/github/hub/releases/tag/v2.3.0), the same time `hub issue` was introduced so this won't be a breaking change due to hub versions as the current CLU implementation requires at least `2.3.0` as well.

Closes #1.

danielbachhuber commented on src/Runner.php:
```diff
@@ -172,7 +172,7 @@ public function start( $target_dir, $opts ) {
 
 	private function checkExisting() {
-		exec('hub issue', $output_lines, $return_code);
+		exec('hub pr list hub pr list --format="%t%n" --state=open', $output_lines, $return_code);
```
```suggestion
		exec('hub pr list --format="%t%n" --state=open', $output_lines, $return_code);
```

ataylorme replied:
whoops 😬

greg-1-anderson commented:
I have not tested but I am 👍 on the concept. LMK if you want me to test it before you merge.

ataylorme commented:
@greg-1-anderson I just tested locally with this branch of `clu` on `example-wordpress-composer`, [which has been getting a new CLU PR daily](https://github.com/pantheon-systems/example-wordpress-composer/pulls?q=is%3Apr+sort%3Aupdated-desc+is%3Aclosed) when the CircleCI job on cron runs.

The first time I ran it I saw

```sh
andrewtaylor:example-wordpress-composer/ (master) $ clu                                                                                                
Found required executables on system: git, composer, hub
Changed into directory: /Users/andrewtaylor/Development/Pantheon/example-wordpress-composer

composer install --no-dev --no-interaction
...
https://github.com/pantheon-systems/example-wordpress-composer/pull/122
Success: Created pull request with composer.lock changes.
```

and a new PR was opened

the second time I ran it I saw

```sh
andrewtaylor:example-wordpress-composer/ (master) $ clu
Found required executables on system: git, composer, hub
Changed into directory: /Users/andrewtaylor/Development/Pantheon/example-wordpress-composer

Using existing branch: clu-2019-05-24-16-20
Switched to branch 'clu-2019-05-24-16-20'
```

and no new PR was created, rather the existing branch was used.

ataylorme commented:
@danielbachhuber once this is merged are you okay with a minor release? I'd like to get out CI processes updated with this fix as the daily PRs are noisy

danielbachhuber commented:
> @danielbachhuber once this is merged are you okay with a minor release? I'd like to get out CI processes updated with this fix as the daily PRs are noisy

Yep! Done with https://github.com/danielbachhuber/composer-lock-updater/releases/tag/v0.5.0

Thanks for the pull request :)
